apps/moderation/serializers.py

```diff
@@ -1,22 +1,66 @@
 from rest_framework import serializers
-from rest_framework.serializers import ModelSerializer
 
+common_instant_search_fields = {
+    'attributees': {'model': 'entities.entity'},
+    'related_entities': {'model': 'entities.entity'},
+    'tags': {'model': 'topics.topic'},
+    'location': {'model': 'places.place'},
+    'images': {'model': 'images.image'},
+}
 
-def get_moderated_model_serializer(
-    model_serializer: type[ModelSerializer],
-) -> type[ModelSerializer]:
-    """Return the serializer to be used for moderation of a moderated model.
-    Adds 'fields' field to given model_serializer.
-    """
 
-    class ModeratedModelSerializer(model_serializer):
-        fields = serializers.ReadOnlyField(source='get_moderated_fields')
+class ModeratedModelSerializer(serializers.ModelSerializer):
+    moderated_fields_excludes = ['id', 'meta', 'admin_url', 'absolute_url', 'cached_tags']
+
+    instant_search_fields = {}
+
+    def get_instant_search_fields(self) -> dict:
+        return common_instant_search_fields | self.instant_search_fields
+
+    def get_instant_search_field(self, field_name: str):
+        field = self.get_instant_search_fields().get(field_name)
+        if field:
+            field_info = {'model': field['model']}
+            if 'filters' in field:
+                field_info['filters'] = field['filters']
+            return field_info
+        return None
+
+    def get_choices_for_field(self, field, field_name: str):
+        if field_name in self.get_instant_search_fields():
+            return None
+        return getattr(field, 'choices', None)
+
+    def get_moderated_fields(self) -> list[dict]:
+        """
+        Return a serialized list of the model's moderated fields.
+
+        This can be used to construct forms intelligently in front-end code.
+        """
+        fields = []
+        field: serializers.Field
 
-        class Meta(model_serializer.Meta):
-            fields = model_serializer.Meta.fields + ['fields']
+        for field_name, field in self.get_fields().items():
+            if field_name not in self.moderated_fields_excludes:
+                data = {
+                    'name': field_name,
+                    'type': field.__class__.__name__,
+                    'editable': not getattr(field, 'read_only', False),
+                    'required': not getattr(field, 'required', False),
+                    'allow_blank': not getattr(field, 'allow_blank', False),
+                    'verbose_name': getattr(field, 'verbose_name', None),
+                    'help_text': getattr(field, 'help_text', None),
+                    'choices': self.get_choices_for_field(field, field_name),
+                }
+                instant_search_field = self.get_instant_search_field(field_name)
+                if instant_search_field:
+                    data.update(data | {'instant_search': instant_search_field})
+                fields.append(data)
```

Comment:
I think there's a redundant merge here. Maybe we could do 
```py
data.update({'instant_search': instant_search_field})
```
or
```py
data['instant_search'] = instant_search_field
```

Reply:
Second one night work the same, but first one seems like it would replace `data` with just given dict?

Reply:
I think it only replaces overlapping keys, which is the same behavior as the `|` operator.

Reply:
Gotcha, for some reason I assumed `update` would clear & set.